odoo/cli/db.py

```diff
@@ -0,0 +1,162 @@
+                 "or if `-`, dumped to stdout",
+        )
+
+        duplicate = subs.add_parser("duplicate", help="Duplicate a database including filestore.")
+        duplicate.set_defaults(func=self.duplicate)
+        duplicate.add_argument(
+            '-f', '--force', action='store_const', default=False, const=True,
+            help="delete `target` database before copying if it exists"
+        )
+        duplicate.add_argument("source")
+        duplicate.add_argument("target", help="database to copy `source` to, must not exist unless `-f` is specified in which case it will be dropped first")
+
+        rename = subs.add_parser("rename", help="Rename a database including filestore.")
+        rename.set_defaults(func=self.rename)
+        rename.add_argument(
+            '-f', '--force', action='store_const', default=False, const=True,
+            help="delete `target` database before renaming if it exists"
+        )
+        rename.add_argument('source')
+        rename.add_argument("target", help="database to rename `source` to, must not exist unless `-f` is specified, in which case it will be dropped first")
+
+        drop = subs.add_parser("drop", help="Delete a database including filestore")
+        drop.set_defaults(func=self.drop)
+        drop.add_argument("database", help="database to delete")
+
+        args = parser.parse_args(cmdargs)
+
+        config.parse_config([
+            val
+            for k, v in vars(args).items()
+            if v is not None
+            if k in ['config', 'data_dir', 'addons_path'] or k.startswith(('db_', 'pg_'))
+            for val in [
+                '--data-dir' if k == 'data_dir'\
+                    else '--addons-path' if k == 'addons_path'\
+                    else f'--{k}',
+                v,
+            ]
+        ])
+        # force db management active to bypass check when only a
+        # `check_db_management_enabled` version is available.
+        config['list_db'] = True
+        report_configuration()
+
+        args.func(args)
+
+    def load(self, args):
+        db_name = args.database or Path(args.dump_file).stem
+        self._check_target(db_name, delete_if_exists=args.force)
+
+        url = urllib.parse.urlparse(args.dump_file)
+        if url.scheme:
+            stderr.write(f"Fetching {args.dump_file}...")
+            r = requests.get(args.dump_file, timeout=10)
+            if not r.ok:
+                exit(f"Unable to fetch {args.dump_file}: {r.reason}")
+
+            stderr.write(" done\n")
+            dump_file = io.BytesIO(r.content)
+        else:
+            stderr.write(f"Restoring {args.dump_file}...\n")
+            dump_file = args.dump_file
+
+        if not zipfile.is_zipfile(dump_file):
+            exit("Not a zipped dump file, use `pg_restore` to restore raw dumps,"
+                 " and `psql` to execute sql dumps or scripts.")
+
+        restore_db(db=db_name, dump_file=dump_file, copy=True)
```

Comment:
This part is a little fragile because of the loading of the registry. We may have noise or even crash trying
- to download a enterprise database without giving the correct addons_path
- to download a 13.0 database (will crash)

The copy option may be needed but not always, we may want to add an option for that.

Finding the filestore using the registry looks useless (we can check but I can't find why this is needed... )

We may change slightly the restore_db to find the filestore using the config  and not load the registry if we don't have the copy option. This would allow us to use this instead of the odooget script.